YAXLib/StringUtils.cs

```diff
@@ -1,334 +1,369 @@
+            if (string.IsNullOrEmpty(elemAddr))
+                return ".";
+
+            // replace all back-slaches to slash
+            elemAddr = elemAddr.Replace('\\', '/');
+
+            var sb = new StringBuilder(elemAddr.Length);
+            var steps = elemAddr.SplitPathNamespaceSafe();
+            foreach(var step in steps)
+            {
+                sb.Append("/" + RefineSingleElement(step));
+            }
+
+            return sb.Remove(0, 1).ToString();
+        }
+
+        /// <summary>
+        /// Heuristically determines if the supplied name conforms to the "expanded XML name" form supported by the System.Xml.Linq.XName class.
+        /// </summary>
+        /// <param name="name">The name to be examined.</param>
+        /// <returns><c>true</c> if the supplied name appears to be in expanded form, otherwise <c>false</c>.</returns>
+        public static bool LooksLikeExpandedXName(string name)
+        {
+            // XName permits strings of the form '{namespace}localname'. Detecting such cases allows
+            // YAXLib to support explicit namespace use.
+            // http://msdn.microsoft.com/en-us/library/system.xml.linq.xname.aspx
+
+            name = name.Trim();
+
+            // Needs at least 3 chars ('{}a' is, in theory, valid), must start with '{', must have a following closing '}' which must not be last char.
+            if (name.Length >= 3)
+            {
+                if (name[ 0 ] == '{')
+                {
+                    int closingBrace = name.IndexOf('}', 1);
+                    return closingBrace != -1 && closingBrace < (name.Length - 1);
+                }
+            }
+            return false;
+        }
+
+        
+        /// <summary>
+        /// Refines a single element name. Refines the location string. Trims it, and replaces invlalid characters with underscore.
+        /// </summary>
+        /// <param name="elemName">Name of the element.</param>
+        /// <returns>the refined element name</returns>
+        public static string RefineSingleElement(string elemName)
+        {
+            if (elemName == null)
+                return null;
+
+            elemName = elemName.Trim(' ', '\t', '\r', '\n', '\v', '/', '\\');
+            if (IsSingleLocationGeneric(elemName))
+            {
+                return elemName;
+            }
+            else if (LooksLikeExpandedXName(elemName))
+            {
+                // thanks go to CodePlex user: tg73 (http://www.codeplex.com/site/users/view/tg73)
+                // for providing the code for expanded xml name support
+
+                // XName permits strings of the form '{namespace}localname'. Detecting such cases allows
+                // YAXLib to support explicit namespace use.
+                // http://msdn.microsoft.com/en-us/library/system.xml.linq.xname.aspx
+
+                // Leave namespace part alone, refine localname part.
+                int closingBrace = elemName.IndexOf( '}' );
```

Comment:
I don't think these changes are needed (removed `var`)

It's poluting the pr

Reply:
Totally agree, it was just a personal preference to change this because I don't like the idea of "hiding" types when they are not explicitly declared both on the left or right hand side and I have to use the IDE to get this information.
To resolve this I would have to redo the all of this from a vanilla version of YAXLib. Do we want this?

Reply:
Let discuss the future code base [outside the pr](https://github.com/orgs/YAXLib/teams/maintainers/discussions/7)

Reply:
> To resolve this I would have to redo the all of this from a vanilla version

ReSharper could change to *var syntax style* across the solution.